New Radial Menu

Copern opened the pull request:
- Adds in the new radial menu.
- Adds shortcuts to the main item ring : shift-click to examine, ctrl-click to pull, and click to pick up.
- Menu remains open when the user right clicks on an item. No longer requires the user to hold down the mouse button.
- Clicking anywhere outside of the menu will close it.
- Certain actions will leave the menu open (for example, examine) while others close it.
- If there are less items than there are segments in the radial, the segments will be transparent and inactive. Inactive segments will will not capture click actions and will allow interaction with the world.
- If there are more items than there are segments, scrolling through the items is allowed through either the mouse wheel or by dragging the item ring around.
- No longer opens up items that are underneath the inventory slots.
- The menu and its buttons are no longer destroyed and recreated every time a user opens and closes the menu.

Known Issues and/or concerns
- Codacy has a some issues with the old RightClick* classes, I've kept changes to those to a minimum for now.
- Some actions do not have icons. As a temporary solution, instead of an icon, it will show text describing the action. I'm not a spriter so someone will need to make some icons for them.
- Some items need right click appearance changes and/or sprite renderer reordering. Again, as a temporary solution, text will show instead of an icon in these cases.
- Some bugs that were in the old menu, such as being able to open up the menu over another window, are still there.
- The menu does not currently update the available actions when moving into or out of range of the items. This will be changed later.
- I wanted to add some customization options before this release. However, the options menu is a royal pain to work with so they won't make it this time. Some planned customization options include changing the number of segments, changing the scale, etc.

Copern committed:
Implementation of the new right click radial menu.

Adds in the radial menu proposed by the UI team.
- Adds shortcuts to the main item ring : shift-click to examine, ctrl-click to pull, and click to pick up.
- Menu remains open when the user right clicks on an item. No longer requires the user to hold down the mouse button.
- Items appear faded until a user moves over the button.
- Right clicking will open a branch that angles towards the center of the screen with the menu attached to it. Moving around will update the direction of the branch and menu depending on what quadrant the origin point is currently in.
- Clicking anywhere outside of the menu will close it.
- Certain actions will leave the menu open (for example, examine) while others close it.
- Buttons expand or shrink based on the number of items in a pile. If there are more than 8 items in the stack, then the menu will be scrollable.
- Scrolling is allowed through either the mouse wheel (4 items at a time) or by dragging the item ring around.
- No longer opens up items that are underneath the inventory slots.
- The menu and its buttons are lazily created and are no longer destroyed and recreated every time a user opens and closes the menu.

Copern committed:
Remove old radial menu

Copern committed:
Remove an obsolete struct

Copern committed:
Appease codacy

Copern committed:
Move radial mouse wheel scrolling into its own component.

Also fix unity conveniently forgetting to save some interaction settings.

Copern committed:
And undo gamedata change that slipped through.

Copern committed:
Fix a small bug introduced with wheel scroll component.

Copern committed:
Some organizing

Copern committed:
Small optimization. Only need to calculate the item center once per setup.

Copern committed:
Switch from UnityEvent to Action for OnIndexChanged

UnityEvent took a non-trivial amount of time when changing indices.

Copern committed:
Update AmmoShotgunDart.prefab

Copern committed:
ItemArcAngle is frequently fetched so calculate once per setup. Reorder getters/setters.

Copern committed:
More appropriate name for arc angles.

Copern committed:
Couple small changes

Copern committed:
Prefab fix. Switch to == false.

Copern committed:
Use scrollhandler interface instead of update.

Copern committed:
Fix various visual issues. Disable mouse wheel while dragging. Prevent interactions while dragging.

Copern committed:
Tiny visual adjustment.

Copern committed:
Button wasn't supposed to be serialized.

Copern committed:
Getter property not used so switching to a set method.

Copern committed:
Add in range check.

Copern committed:
Fix a scrolling bug when arc measure is lower than 360.

Copern committed:
Add palette size

Copern committed:
Change player radial background color.

Copern committed:
Radial now always shows 8 segments.

Any segments without items will be mostly transparent and they allow world interaction through them.
Allow interaction through the description background.
Radial menu has a little transparency now.
Radial branch adjusts it size if the action radial covers it to allow radial to be transparent.
Adjusted button background colors.

Copern committed:
Items in the radial now center on the branch line.

For example, if there's only one item it will be closest to the branch line.

Copern committed:
Give the item radial inner ring a little transparency too.

Copern committed:
And give the action radial inner ring some transparency too...

Copern committed:
Fix potential NRE. Use IsInReach.

Copern committed:
Animations and formatting

- Scrolling via scroll wheel now tweens the radial rotation towards the target instead of being instant.
- The first and last item icon now shrink/expand depending on how close/far from the rotation point the icon is.
- Rotation arrow indicators now shrink/expand depending on the direction the radial is rotating.
- Lessened the transparency of item icons.
- Consolidated and moved some things around
- Formatting because I'm a derp and forgot to adjust some IDE settings after an update (what an ugly mix of spaces and tabs)

Copern committed:
That wasn't supposed to be there...

Copern committed:
Potentially fix a missing reference exception by explicitly passing the gameobject.

Copern committed:
Remove delegate caching for now.

Copern committed:
Various fixes and adjustments.

Remove icon fading/greying all together. Replace it with button highlighting/selection.
Some actions do not currently have an icon. As a temporary solution, rather than display a question mark icon, show a label that describes the action.
Fix an issue with radial rotation.
Fix buttons receiving raycasts when they were disabled.
Fix LeanTween sometimes breaking due to the LeanTween object not being created because the tweens array already existed under some circumstances.

Copern committed:
Fix minor radial highlighting issues. Pull a local function out to an instance method.

Copern committed:
Button backup label truncate instead of overflow.

corp-0 commented on UnityProject/Assets/Scripts/UI/Core/Animations/AnimatedRadialRotation.cs:
```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using UI.Core.Radial;
+using UnityEngine;
+
```
add a namespace for any new script you add. In this case the namespace would be ``UI.Core.Animations``

corp-0 commented on UnityProject/Assets/Scripts/UI/Core/Animations/AnimationDirection.cs:
```diff
@@ -0,0 +1,6 @@
+﻿public enum AnimationDirection
```
same here, namespace.

corp-0 commented on UnityProject/Assets/Scripts/UI/Core/Animations/ReversibleObjectScale.cs:
```diff
@@ -0,0 +1,76 @@
+﻿using UnityEngine;
+
```
namespace

Copern committed:
Various changes

Namespacing.
Use the AirLockAnimator to fetch sprites for some doors for now.
Add an outline to the item description.
Slightly less transparent buttons.
Add a RightClickAppearence for hydroponics trays.
Remove redundant/unused RightClickAppearence from various prefabs.

Copern committed:
Codacy Appeasement
